sync-core/src/main/java/com/cloudant/sync/datastore/PreparedAttachment.java

```diff
@@ -43,20 +47,47 @@ public PreparedAttachment(Attachment attachment,
                               String attachmentsDir) throws AttachmentException {
         this.attachment = attachment;
         this.tempFile = new File(attachmentsDir, "temp" + UUID.randomUUID());
-        FileInputStream tempFileIS = null;
+        InputStream attachmentInStream = null;
+        OutputStream tempFileOutStream = null;
+        MessageDigest sha1 = null;
         try {
-            FileUtils.copyInputStreamToFile(attachment.getInputStream(), tempFile);
-            this.sha1 = Misc.getSha1((tempFileIS = new FileInputStream(tempFile)));
+            attachmentInStream = attachment.getInputStream();
+
+            //Use FileUtils to create folder structure and file if necessary for output stream
+            tempFileOutStream = FileUtils.openOutputStream(this.tempFile);
+
+            sha1 = MessageDigest.getInstance("SHA-1");
+            int bufSiz = 1024;
+            byte buf[] = new byte[bufSiz];
```

Comment:
Can be both ways, but `byte[] buffer` is less confusing and the recommended way.  Fixed: db71913691e9202f5420a3af5cae52f530b05fc0